addons/binding/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/SensorThingHandler.java

```diff
@@ -276,7 +276,7 @@ public void valueUpdated(String channelID, SensorState state) {
                 }
                 break;
             case BindingConstants.CHANNEL_BUTTONEVENT:
-                if (buttonevent != null) {
+                if (buttonevent != null && !initializing) {
```

Comment:
I suggest to check against the `last_updated` value (see https://github.com/eclipse/smarthome/pull/6770) because I am faceing a lot more trigger - nearly every 5 min, w/o the button being pressed:
```
2019-01-04 22:50:12.792 [INFO ] [clipse.smarthome.model.script.deconz] - Trigger für TRÅDFRI Fernbedienung button pressed: 'deconz:switch:XXX:2:buttonevent triggered 4003'
2019-01-04 22:55:13.012 [INFO ] [clipse.smarthome.model.script.deconz] - Trigger für TRÅDFRI Fernbedienung button pressed: 'deconz:switch:XXX:2:buttonevent triggered 4003'
2019-01-04 23:00:12.853 [INFO ] [clipse.smarthome.model.script.deconz] - Trigger für TRÅDFRI Fernbedienung button pressed: 'deconz:switch:XXX:2:buttonevent triggered 4003'
2019-01-04 23:05:12.969 [INFO ] [clipse.smarthome.model.script.deconz] - Trigger für TRÅDFRI Fernbedienung button pressed: 'deconz:switch:XXX:2:buttonevent triggered 4003'
```

Reply:
This exactly 5 minutes happen due to a timeout every 5 minutes on the websocket connection to the deconz rest api and is not related to pulling.
As I didn't find where the exact time for the timeout is comming from I'd have to use magic numbers for fixing this for now.
In my eyes a flag checking whether we entered from a regular call or from changing the bridge state is currently the cleanest solution to avoid the repeating trigger.

Reply:
We should watch the deconz web UI. They probably send a keep-alive to the websocket connection every now and then.

Reply:
Okay, I see. I expected it to be related to the socket timeout. I will give it a go and update my installed version to include your fix.

Reply:
Ok, guys I took a somehow deep dive into the phoscon web app behaviour, the rest-plugin sources, and the current state of this binding the last couple of hours and here  are my findings:

## 5-Minute timeout on websocket
I still didn't find where this is set, but may be a default in the Qt-Sockets, so for the current state this is the best fix I can provide quickly.

## Why web app gets no timeout
Actually the web app has scheduled polling of all states (groups, lights, config, sensors) on an average of every 15 seconds. Which seems to force deconz to keep the clients websocket running.
Also the api-decription [here](http://dresden-elektronik.github.io/deconz-rest-doc/polling/) mentions the expectation of polls. I guess they expect people use their API completely and not only the sensors part.

## So whats next?
On the path to next stable release, I think a rewrite including polls and also event triggers for all sensors should be done. I would start with it right away and make a new PR replacing this one then, as it should become obsolete.

As the polling actually needs the basic rest implementation already, I'd suggest implementing the lights in a second step, too, so people without a hue bridge at home can drive their conbee/raspbee setup with only the deconz-binding, which is expected behaviour in my eyes.

### what do you think?